ofrak_core/test_ofrak/components/test_cpio_component.py

```diff
@@ -46,12 +46,9 @@ async def repack(self, cpio_resource: Resource) -> None:
         await cpio_resource.pack_recursively()
 
     async def verify(self, repacked_cpio_resource: Resource) -> None:
-        resource_data = await repacked_cpio_resource.get_data()
-        with tempfile.NamedTemporaryFile() as temp_file:
-            temp_file.write(resource_data)
-            temp_file.flush()
+        async with repacked_cpio_resource.temp_to_disk() as temp_path:
             with tempfile.TemporaryDirectory() as temp_flush_dir:
-                command = f"(cd {temp_flush_dir} && cpio -id < {temp_file.name})"
+                command = f"(cd {temp_flush_dir} && cpio -id < {temp_path})"
                 subprocess.run(command, check=True, capture_output=True, shell=True)
```

Comment:
This should be converted to use a list of arguments instead of a single string. I know you're keeping it consistent with the way it was done before, but there is much lower risk of command injection if we change `cwd` via the `subprocess.run` call instead of with `cd` in a subshell. 

Moreover, this should use the `asyncio` subprocess call instead of a blocking one.